plugins/oracle_database_logs.yaml

```diff
@@ -0,0 +1,209 @@
+      attributes:
+        plugin_id: oracledb
+      operators:
+        - id: listener_router
+          type: router
+          routes:
+            - output: xml_listener_regex_parser
+              expr: body matches '<msg time=\\D\\d{4}-\\d{2}-\\d{2}T\\d{1,}:\\d{2}:\\d{2}.\\d+[+-]\\d+:\\d+'
+            - output: linux_listener_regex_parser
+              expr: body matches '/[\\w\\d/]+:'
+            - output: windows_listener_regex_parser
+              expr: body matches '\\\\[\\d\\D\\\\]*:'
+            - output: trace_listener_regex_parser
+              expr: body matches '\\d{4}-\\d{2}-\\d{2}T\\d{1}:\\d{2}:\\d{2}.\\d+[+-]\\d+:\\d+'
+          default: add
+
+        - id: linux_listener_regex_parser
+          type: regex_parser
+          regex: '(?P<path>/[\w\d/\\]+:)\s(?P<status_code>\w+).*\n\nLSNRCTL for (?P<os>[\w\s]+):\sVersion\s(?P<listener_version>[\d\.]+)\s-\s\w+\son\s(?P<timestamp>\d{2}-\w{3}-\d{4}\s\d{2}:\d{2}:\d{2})\n\n(?P<copyright>[\d\w[:ascii:]]+)\n\n(?P<message>[\d\w[:ascii:]]+)'
+          timestamp:
+            parse_from: attributes.timestamp
+            layout: '%d-%h-%Y %T'
+          output: add
+
+        - id: windows_listener_regex_parser
+          type: regex_parser
+          regex: '(?P<path>\\[\d\w\\]+:)\s(?P<status_code>\w+).*\n\nLSNRCTL for (?P<os>[\w\s]+):\sVersion\s(?P<listener_version>[\d\.]+)\s-\s\w+\son\s(?P<timestamp>\d{2}-\w{3}-\d{4}\s\d{2}:\d{2}:\d{2})\n\n(?P<copyright>[\d\w[:ascii:]]+)\n\n(?P<message>[\d\w[:ascii:]]+)'
+          timestamp:
+            parse_from: attributes.timestamp
+            layout: '%d-%h-%Y %T'
+          output: add
+
+        - id: trace_listener_regex_parser
+          type: regex_parser
+          regex: '(?P<timestamp>\d{4}-\d{2}-\d{2}T\d{1}:\d{2}:\d{2}.\d+[+-]\d+:\d+)\s+(?P<message>[\d\D\s]+)'
+          timestamp:
+            parse_from: attributes.timestamp
+            layout: '%Y-%m-%dT%T.%L%j'
+          output: add
+
+        - id: xml_listener_regex_parser
+          type: regex_parser
+          regex: '<msg\s+time=\D(?P<timestamp>\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d{3}[+-]\d{2}:\d{2})\D(\s+)?(org_id=\D(?P<org_id>[\w\d]+)\D)?(\s+)?(comp_id=\D(?P<comp_id>[\w\d]+)\D)?(\s+)?(msg_id=\D(?P<msg_id>[\w\d:]+)\D)?(\s+)?(type=\D(?P<type>\w+)\D)?(\s+)?(group=\D(?P<group>[\w ]+)\D)?(\s+)?(level=\D(?P<level>\d+)\D)?(\s+)?(host_id=\D(?P<host_id>[\.\w\d-_]+)\D)?(\s+)?(host_addr=\D(?P<host_addr>[:a-fA-F\d\.]+)\D)?(\s+)?(module=\D(?P<module>[\w-\s@()]+)\D)?(\s+)?(pid=\D(?P<pid>\d+)\D)?(\s+)?(\s+)?(version=\D(?P<version>\d+)\D)?(\s+)?>\s+<txt>(?P<txt>[\d\D\s]+)</txt>(\s+)?</msg>'
+          timestamp:
+            parse_from: attributes.timestamp
+            layout: '%Y-%m-%dT%T.%L%j'
+          output: add
+
+        - type: add
+          field: attributes.log_type
+          value: oracledb.listener
+    {{ end }}
+
+  service:
+    pipelines:
+      logs:
+        receivers:
+          - filelog/audit
+          - filelog/alert
+          - filelog/listener
```

Comment:
I think these need conditionals based on the enable parameters (e.g. if enable_listener_log is false, filelog/listener shouldn't be included here)

Reply:
Good call